lib/financial_period.rb

```diff
@@ -0,0 +1,24 @@
+module FinancialPeriod
+  def self.quarter_from_date(date)
+    month = date.month
+    case month
+    when 1, 2, 3
+      "4"
+    when 4, 5, 6
+      "1"
+    when 7, 8, 9
+      "2"
+    when 10, 11, 12
+      "3"
+    end
+  end
+
+  def self.year_from_date(date)
+    month = date.month
+    year = date.year
+    if (4..12).cover?(month)
```

Comment:
TIL about `.cover?` method. Didn't come across it before (very useful!). I like the way you implemented the logic to get the year of the forecast :-)

Reply:
It was new to me as well - I spent far too long trying to craft these methods and I still find them a bit odd to look at, but they do get the job done!

Reply:
I think they look elegant!